app/services/sendgrid_client.rb

```diff
@@ -0,0 +1,62 @@
+class SendgridClient
+  TIMEOUT = 2 # seconds
+
+  class Error < StandardError; end
+
+  attr_reader :api_key, :api_user
+
+  def initialize(api_key:, api_user:, http_opts: {})
+    timeout = http_opts.fetch(:timeout, TIMEOUT)
+    persistent = http_opts.fetch(:persistent, false)
+
+    @connection ||= Excon.new('https://api.sendgrid.com',
+      persistent: persistent,
+      connect_timeout: timeout)
+
+    @api_key = api_key
+    @api_user = api_user
+  end
+
+  def get_request(endpoint, query, timeout: TIMEOUT)
+    fail Error, 'no credentials set' if credentials_missing?
```

Comment:
This is to keep backwards compatibility. The current behaviour in `master` allows to run development without Sendgrid, and the rest of the test suite doesn't expect any API calls out to Sendgrid.

I'd like to refactor in a different PR so that Sendgrid is explicitly disabled in development and in test rework the specs so that it allows Sendgrid API calls.